machine-learning/05-machine_learning_exercise.Rmd

```diff
@@ -0,0 +1,245 @@
+
+### Is there a latent variable associated with the proteasome pathway?
+
+We can use a chain of `dplyr` functions to filter `summary_df`.
+Let's call `FDR < 0.05` associated; this is the same cutoff we used in the
+PLIER notebook earlier.
+
+```{r}
+summary_df %>%
+  dplyr::filter(,  # significant associations only
+                # grepl returns TRUE when the pattern "proteasome" is found
+                # in the pathway column
+                grepl("proteasome", pathway, ignore.case = TRUE))
+```
+
+The LV index column of `summary_df` tells us _which row_ in the LV by sample
+matrix `plier_results$B` we should look at.
+To make plotting easier downstream, let's make a data.frame that contains two 
+columns, `geo_accession` and `proteasome_lv` that contain the sample identifiers
+and the values for the latent variable associated with the proteasome gene set, 
+respectively.
+Recall that the column names of `plier_results$B` are the sample identifiers.
+(Hint: `?colnames`)
+
+Call this data.frame `proteasome_df`.
+
+```{r}
+proteasome_df <- data.frame(geo_accession = ,
+                            proteasome_lv = )
+```
+
+Now that you have `proteasome_df`, let's join it with our metadata using
+an `inner_join` from `dplyr` to form a new data.frame called `prot_meta_df`.
+We'll join by the sample names, which is what `by = "geo_accession"` specifies.
+
+```{r}
+prot_meta_df <- dplyr::inner_join(x = , 
+                                  y = , 
+                                  by = "geo_accession")
+```
+
+Let's find out what samples have the highest expression values for the
+proteasome latent variable.
+We can sort `prot_meta_df` with `arrange`.
+Use the `Help` panel or `?arrange` to learn what to expect from this function.
+
+```{r}
+prot_meta_df %>%
+```
+
+What do you notice about the samples with the highest values?
+Look up the mechanism of action for the drugs in `treatment`.
+Do these results make sense?
+
+### Line graph
+
+Let's make a plot that shows the change of the proteasome latent variable
+expression level over time, when treated with the drug that comes to the top
+of the list.
+We'll look at a single tissue of your choosing.
+This requires use to do the following:
+
+1. Filter `prot_meta_df` to samples from your tissue of interest and the
+   drug that we are interested in
+2. Make a line graph that tracks the expression value of the proteasome 
+   latent variable during different time points of the experiment.
+
+Save this plot as `p`.
+
+```{r}
+p <- prot_meta_df %>%
+  # use these lines to filter to only the samples that are treated with the drug
+  # of interest and to the tissue of your choice
+  %>% 
+  ggplot(aes(x = ,  # what needs to be on the x-axis?
+             y = ,  # what needs to be on the y-axis?
+             color = cell_line,  # color by individual cell line
+             # we need to use group to make our line plot
+             group = cell_line)) +
+  # use facets to separate dosages
```

Comment:
We’ve had them use `facet_wrap` in the scRNAseq exercise. So hopefully if they reached that part of the exercise they remember how to do it.